src/client/render/render-core.js

```diff
@@ -138,114 +139,28 @@ let renderCore = function() {
 			}
 		},
 
-		getDefaultModel: function() {
-			return _defaultModel;
+		clientUpdateCallback: function() {
+			// Overridable function for the client and editor to attach their update functions to.
 		},
 
-		getMainScene: function() {
-			return _mainScene;
+		waterReflectsLevel: function() {
+			return config.graphics.reflection.level;
 		},
 
-		updateMarbles: function() {},
-
-		addMarbleMesh: function(data) {
-			let marbleMesh = new MarbleMesh(data);
-			_marbleMeshes.push(marbleMesh);
-			_mainScene.add(marbleMesh.mesh);
-			_mainScene.add(marbleMesh.nameSprite);
+		waterReflectsMarbles: function() {
+			return config.graphics.reflection.marbles;
 		},
 
-		removeAllMarbleMeshes: function() {
-			for (let marble of _marbleMeshes) {
-				for (let i = marble.mesh.children.length; i >= 0; i--) {
-					_mainScene.remove(marble.mesh.children[i]);
-				}
-				_mainScene.remove(marble.nameSprite);
-				_mainScene.remove(marble.mesh);
-			}
-
-			_marbleMeshes = [];
+		getDefaultModel: function() {
+			return _defaultModel;
 		},
 
-		updateMarbleMeshes: function(newPositions, newRotations, delta) {
-			for (let i = 0; i < _marbleMeshes.length; i++) {
-				// Positions
-				_marbleMeshes[i].mesh.position.x = THREE.Math.lerp(_marbleMeshes[i].mesh.position.x || 0, newPositions[i * 3 + 0], delta);
-				_marbleMeshes[i].mesh.position.y = THREE.Math.lerp(_marbleMeshes[i].mesh.position.y || 0, newPositions[i * 3 + 2], delta);
-				_marbleMeshes[i].mesh.position.z = THREE.Math.lerp(_marbleMeshes[i].mesh.position.z || 0, newPositions[i * 3 + 1], delta);
-
-				// Rotations
-				_marbleMeshes[i].mesh.quaternion.set(
-					newRotations[i * 4 + 0],
-					newRotations[i * 4 + 1],
-					newRotations[i * 4 + 2],
-					newRotations[i * 4 + 3]
-				);
-
-				// Also update the nameSprite position
-				if (_marbleMeshes[i].nameSprite) {
-					_marbleMeshes[i].nameSprite.position.x = (_marbleMeshes[i].mesh.position.x || 0);
-					_marbleMeshes[i].nameSprite.position.y = (_marbleMeshes[i].mesh.position.y || 0) + _marbleMeshes[i].size - .1;
-					_marbleMeshes[i].nameSprite.position.z = (_marbleMeshes[i].mesh.position.z || 0);
-				}
-			}
+		getMainScene: function() {
```

Comment:
I agree there's no need for it for `getMainScene`, so I'll change that. The default model had a getter because at one point it might've involved a Promise due to it loading dynamically, but that's been solved a different way for now.